app/uk/gov/hmrc/teamsandrepositories/models/BuildJob.scala

```diff
@@ -22,54 +22,96 @@ import uk.gov.hmrc.mongo.play.json.formats.MongoJavatimeFormats
 
 import java.time.Instant
 
-case class BuildJob(
-  service    : String,
-  jenkinsURL : String,
-  latestBuild: Option[BuildJobBuildData]
-)
+case class BuildData(
+                      number: Int,
+                      url: String,
+                      timestamp: Instant,
+                      result: Option[BuildResult]
+                    )
+object BuildData {
+
+  val mongoFormat: OFormat[BuildData] =
+    ((__ \ "number").format[Int]
+      ~ (__ \ "url").format[String]
+      ~ (__ \ "timestamp").format(MongoJavatimeFormats.instantFormat)
+      ~ (__ \ "result").formatNullable[BuildResult]
+      ) (apply, unlift(unapply))
+
+  val apiWrites: Writes[BuildData] =
+    ((__ \ "number").write[Int]
+      ~ (__ \ "url").write[String]
+      ~ (__ \ "timestamp").write[Instant]
+      ~ (__ \ "result").writeNullable[BuildResult]
+      ) (unlift(unapply))
+
+  val jenkinsReads: Reads[BuildData] =
+    ((__ \ "number").read[Int]
+      ~ (__ \ "url").read[String]
+      ~ (__ \ "timestamp").read[Instant]
+      ~ (__ \ "result").readNullable[BuildResult]
+      ) (apply _)
+}
+
+
+sealed trait JenkinsObject
+
+case class JenkinsFolder(service: String, jenkinsURL: String, objects: Seq[JenkinsObject]) extends JenkinsObject
```

Comment:
the instances of `JenkinsObject` should probably be inside the `JenkinsObject` companion object. Once we have the `JenkinsObject` namespace, we could probably remove the `Jenkins` prefix from Folder (which aligns with `BuildJob` and `PipelineJob` (otherwise `BuildJob` should probably have the `Jenkins` prefix too)